src/Compilers/Core/MSBuildTask/GenerateMSBuildEditorConfig.cs

```diff
@@ -101,6 +101,35 @@ public override bool Execute()
             return true;
         }
 
+        // Filenames with special characters like '#' and'{' get written
+        // into the section names in the resulting .editorconfig file. Later,
+        // when the file is parsed in configuration options these special
+        // characters are interpretted as invalid values and ignored by the
+        // processor. We encode the special characters in these strings
+        // before writing them here.
+
+        private static string EncodeString(string p)
+        {
+            StringBuilder builder = new StringBuilder();
+            for (var i = 0; i < p.Length; i++)
+            {
+                builder.Append(p[i] switch
+                {
+                    '*' => "\\*",
+                    '?' => "\\?",
```

Comment:
In Windows, these aren't valid in file names. If it's the same in Linux and macOS, then this won't be reachable.

Reply:
They look valid on Linux. Ignore the previous comment.